### ISSUE TITLE
`no-any-explicit` false positive with type parameter default `Any` 

### ISSUE THREAD

**Issue by jorenham** [KotlinIsland/basedmypy#861]
### Describe the problem

There's no explicit `Any` here:

```py
a = slice(None)
b: slice = a
```

But the following error is reported:

```
Explicit "Any" is not allowed (no-any-explicit) [Ln 2, Col 1]
```

I'm guessing this is because the type parameters of `slice` default to `Any`. But I don't see why such an *implicit* `Any` should be considered as *explicit*.

### Gist to reproduce

```python
a = slice(None)
b: slice = a
```

### Severity

annoying but workaround is available

### Your Environment

basedmypy 2.9.1 (compiled: yes)
Based on mypy 1.14.1

**Comment by jorenham** [comment]
And the weirdest thing is that in this example, it occurs twice:

```py
from typing import Generic, TypeVar

T = TypeVar("T", bound=int | slice)  # E: no-any-explicit

class A(Generic[T]): ...  # <no error>
class B(A[T]): ...  # E: no-any-explicit
```

So with `A` it doesn't report an error, but with `B` it does?

**Comment by jorenham** [comment]
For a real life example, see e.g. [optype/numpy/_to.py](https://github.com/jorenham/optype/blob/08188783144a22c6eb6cdc752c3db9522bda9fbe/optype/numpy/_to.py), which doesn't even import `Any`, yet there are 62 `no-any-explicit` errors reported,  forcing me to use `# mypy: disable-error-code="no-any-explicit"`.

**Comment by KotlinIsland** [comment]
> I'm guessing this is because the type parameters of `slice` default to `Any`. But I don't see why such an _implicit_ `Any` should be considered as _explicit_.

so this is more about understandability? or would you not want an error here at all?

**Comment by jorenham** [comment]
I don't think it's useful in this case, because the `Any` is out of your control. I also don't think that it's useful to report the same error about the same `Any` on multiple occasions. Given the name of the error, I expected it to only be reported once I explicitly write "`Any`",  so that I can actually *see* that it's there. 

I might be wrong here, but I thought that that's also the way that bpr's [`reportExplicitAny`](https://docs.basedpyright.com/latest/benefits-over-pyright/new-diagnostic-rules/#reportexplicitany) works 🤷🏻
